src/bapsiclesession/state.ts

```diff
@@ -45,19 +52,31 @@ const sessionState = createSlice({
 
 export default sessionState.reducer;
 
-export const { setServerState } = sessionState.actions;
+export const { setServerState, setServerName } = sessionState.actions;
 
 export const getCurrentServer = (): AppThunk => async (dispatch, getState) => {
   return getState().bapsSession.currentServer;
 };
 
 export const getServer = (): AppThunk => async (dispatch) => {
-  // TODO Server Details Configurable
+  // Since BAPS Presenter is served by the BAPSicle web server, use the current window path unless custom defined.
   let bapsServer: bapsServer = {
-    hostname: window.location.hostname,
-    port: 13501,
+    hostname: process.env.REACT_APP_BAPSICLE_HOST
+      ? process.env.REACT_APP_BAPSICLE_HOST
+      : window.location.hostname,
+    ws_port: process.env.REACT_APP_WEBSOCKET_PORT
+      ? parseInt(process.env.REACT_APP_WEBSOCKET_PORT)
+      : 13501,
+    ui_protocol: process.env.REACT_APP_BAPSICLE_PROTOCOL
+      ? process.env.REACT_APP_BAPSICLE_PROTOCOL
+      : "http",
+    ui_port: process.env.REACT_APP_BAPSICLE_PORT
+      ? parseInt(process.env.REACT_APP_BAPSICLE_PORT)
+      : parseInt(window.location.port),
```

Comment:
Pretty sure these can be simplified using `??` (e.g. `process.env.REACT_APP_BAPSICLE_HOST ?? window.location.hostname`)

Reply:
Cheers :) Done.